src/cmds/hardware/pin/pin.c

```diff
@@ -144,63 +65,38 @@ int main(int argc, char **argv) {
 		}
 	}
 	
-	if (argc < 2) {
+	if (argc < 4) {
 		print_usage();
 		return 0;
 	}
-
 	opt = 1;
-	if (strlen(argv[opt]) > 1) {
-		// [char]something
-		gpio = gpio_by_name(argv[opt], &pin);
-		if (gpio) {
-			if (pin != PIN_UNSPECIFIED) {
-				int level = extract_level(opt+1, argc, argv);
-				apply(gpio, pin, level);
-				return 0;
-			} else {
-				fprintf(stderr, "Can't extract pin index.");
-				exit(1);
-			}
-		} else {
-			fprintf(stderr, "Can't extract GPIO.");
-			return 0;
-		}
-	} else {
-		gpio = gpio_by_name(argv[opt], &pin);
-		if (gpio) {
-			opt++;
-			if (opt >= argc) {
-				fprintf(stderr, "Unsufficient arguments.");
-				return 0;
-			}
-			pin = pin_by_name(argv[opt]);
-			if (pin != PIN_UNSPECIFIED) {
-				int level = extract_level(opt+1, argc, argv);
-				apply(gpio, pin, level);
-				return 0;
-			} else {
-				fprintf(stderr, "Can't extract pin index.");
-				exit(1);
-			}
-		} else {
-			fprintf(stderr, "Can't extract GPIO.");
-			return 0;
-		}
-	}
 
-	opt++;
-	if (opt >= argc) {
-		fprintf(stderr, "Unsufficient arguments.");
+	gpio = gpio_by_name(argv[opt++]);
+	if (gpio == -1) {
+		fprintf(stderr, "Can't extract GPIO.\n");
 		return 0;
 	}
-	if (PIN_UNSPECIFIED == (pin = pin_by_name(argv[opt]))) {
-		fprintf(stderr, "Pin index is unspecified.");
+
+	if (!sscanf(argv[opt++], "%d", &arg)) {
+		fprintf(stderr, "Pin is not specified.\n");
 		return 0;
 	}
-	opt++;
-	level = extract_level(opt, argc, argv);
-	apply(gpio, pin, level);
+	pin = (1 << arg);
+
+	if (!strcmp("toggle", argv[opt])) {
+		gpio_setup_mode(gpio, pin, GPIO_MODE_OUTPUT);
+		gpio_toggle(gpio, pin);
+	} else {
```

Comment:
May be `} else { if() ... }` -> `} else if () {}` ?

Reply:
Thank you, it looks better. Fixed.